proxy/http2/HTTP2.h

```diff
@@ -62,6 +62,7 @@ const uint32_t HTTP2_INITIAL_WINDOW_SIZE    = 65535;
 const uint32_t HTTP2_MAX_FRAME_SIZE         = 16384;
 const uint32_t HTTP2_HEADER_TABLE_SIZE      = 4096;
 const uint32_t HTTP2_MAX_HEADER_LIST_SIZE   = UINT_MAX;
+const uint32_t HTTP2_MAX_BUFFER_USAGE       = 524288;
```

Comment:
It seems a shame to have yet another magic constant for buffering in HTTP/2. We should look at how the tunnel consumer can provide buffer size feedback to `HttpTunnel`.